lib/analytics/analytics_servce.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:ax_dapp/analytics/event.dart';
+
+class AnalyticsService {
+  AnalyticsService._();
+
+  /// Default instance of the analytics
+  static final instance = AnalyticsService._();
+
+  /// final Tracker? trakingService;
+  /// Loads trackers
+  /// void configureWith(
+  ///   // firebase
+  ///   // aplitude
+  ///   // trakingService
+  ///   // ....
+  /// ) {
+  ///    this.trakingService = trakingService
+  /// }
+
+  ///
+  /// Packages/trackers may have additional lifecycle requirements.
+  /// These methods like reset and other ones should be handled here.
+  ///
+
+  /// Updates tracking services accordingly
+  void track(Event event) {}
```

Comment:
Is the tracking supposed to be done through Google Analytics or something else ?

Reply:
I think the point is to have it independent of any specific provider. I don't know which one will be used.